### DISCUSSION TITLE
Failure in Authentication

### DISCUSSION THREAD

**Discussion by mwatz1234** [tsightler/ring-mqtt#369] [Support Q&A] [answered]
1st time user attempting to setup using the docker method (running on raspberry pi 4 lite).  

I am getting an error message when trying to get the token
    $ docker run -it --rm --mount type=bind,source=/home/pi/ring-mqtt/data,target=/data --entrypoint /app/ring-mqtt/init-ring-mqtt.js tsightler/ring-mqtt
    #
    # Fatal error in , line 0
    # unreachable code
    #
    #
    #
    #FailureMessage Object: 0xbea905ac

Already Completed Steps:
pulled the ring-mqtt docker image (docker pull tsightler/ring-mqtt)
mosquitto docker is setup and HA is configured to it
manually created this folder path /home/pi/ring-mqtt/data
Already setup the Ring Integration

Not sure what to do next for debugging

**Comment by tsightler** [comment, marked as answer]
Looks like you are on Raspbian buster 32-bit?  Due to the outdated docker and libsecomp2 this OS is unable to run many Docker images based on modern Linux distros (ring-mqtt Docker image is based on Alpine 3.15.x).  One option is to upgrade to bullseye, which I believe has addressed these older packages (I don't know this for sure as I don't run 32-bit OS, but I've been told this is addressed in newer release).  IMO, best option is to reinstall with 64-bit OS as long as you have no reason to stay on older 32-bit versions. but that's a reinstall so not painless for sure.

Third, and probably least disruptive option, is to install backported libseccomp2.  There's plenty of instructions around on this, this one seems pretty clear:

https://www.giters.com/SensorsIot/IOTstack/issues/401

**Comment by mwatz1234** [comment]
Thanks for the quick reply.  I backed up the config files and updated to the latest 64-bit OS (raspberry) and the issue got resolved.  Thanks again.
